src/agent/nodes/construction.py

```
@@ -61,13 +63,34 @@ def construction_agent(state: AgentState) -> AgentStateUpdate:
+        agent,
+        local,
+        specs,
+        phase="construction",
+        is_revision=state.is_revision,
+        validation_errors=state.validation_errors,
+    )
+
+    record_phase_trace("construction", collector.export())
+
+    # Back-hop: a missing material layer routes to the material phase.
+    hop = maybe_backhop(result, state, local, "construction")
```

Ruff (0.16.1)

[error] Undefined name `maybe_backhop`

(F821)